src/main/java/com/redhat/cloud/common/clowder/configsource/ClowderConfigSource.java

```diff
@@ -202,6 +205,21 @@ public String getValue(String configKey) {
                         // fall through to fetching the value from application.properties
                 }
             }
+
+            if (configKey.startsWith(CLOWDER_ENDPOINTS)) {
+                JsonArray endpoints = root.getJsonArray("endpoints");
+                if (endpoints == null) {
+                    throw new IllegalStateException("No endpoints section found");
+                }
+                String endpointName = configKey.substring(CLOWDER_ENDPOINTS.length());
+                for (int i = 0; i < endpoints.size(); i++) {
+                    JsonObject endpoint = endpoints.getJsonObject(i);
+                    if (endpoint.getString("name").equals(endpointName)) {
+                        return endpoint.getString("hostname") + ":" + endpoint.getJsonNumber("port").intValue();
```

Comment:
Do we need to do a null check here? getJsonNumber for example could return null if "port" isn't present.

Reply:
Yes it could throw a NPE. Ideally, we should add proper checks for all configuration keys used in `ClowderConfigSource` and throw an exception with a helpful error message each time something is wrong.

Reply:
But this is much larger than this PR so I'd rather do that later 😄

Reply:
While I also don't like throwing a NPE, if the port is not set we can't continue anyway.

Reply:
We have the same issue with the database port BTW (a few lines below).